src/utils.cc

```diff
@@ -61,6 +61,11 @@ std::string escape_for_dot_label(std::string str) {
 	return str;
 }
 
+[[noreturn]] void unreachable() {
+	assert(!"executed unreachable code");
```

Comment:
Should this do something like `__builtin_unreachable();`?

Reply:
It could be an inline function that does
```c++
assert(!"executed unreachable code");
__builtin_trap();
```
Question is, does it matter how this is implemented?

Reply:
But your suggestion kinda does makes me want to use more intrinsics in the codebase, especially `__builtin_ia32_pause()` in the executor loop :P

Reply:
The point is that `__builtin_unreachable()` and its equivalents might transport some information to the compiler which it can use in optimization.

Reply:
Sure, but none of its usages are performance sensitive enough to warrant UB in a release build.